test/functionalTest/cases/2015_notification_templates/notification_templates_with_decoded_chars_in_notifications.test

```diff
@@ -0,0 +1,179 @@
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notification Templates, with decoded chars in notifications
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart
+
+--SHELL--
+
+#
+# 01. Create a subscription with encoded chars in custom payload
+# 02. Create an entity to provoke a notification
+# 03. Dump accumulator to ensure the broker correctly decodes the encoded chars in the custom payload
+# 04. GET the subscription to see the encoded chars in custom payload
+#
+
+echo "01. Create a subscription with encoded chars in custom payload"
+echo "=============================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpExtended": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "payload": "A string with double-quotes: %22${A1}%22",
+      "method":  "POST",
+      "headers": { "hs1": "1", "hs2": "${A1}" },
+      "qs":      { "qs1": "1", "qs2": "${A1}" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+subId=$(echo "$_responseHeaders" | grep Location | awk -F/ '{ print $4 }' | tr -d "\r\n")
+echo
+echo
+
+
+echo "02. Create an entity to provoke a notification"
+echo "=============================================="
+payload='{
+  "id": "E1",
+  "A1": "a1"
+}'
+orionCurl --url /v2/entities?options=keyValues --payload "$payload"
+echo
+echo
+
+
+echo "03. Dump accumulator to ensure the broker correctly decodes the encoded chars in the custom payload"
+echo "==================================================================================================="
+accumulatorDump
+echo
+echo
+
+
+echo "04. GET the subscription to see the encoded chars in custom payload"
+echo "==================================================================="
+orionCurl --url /v2/subscriptions/$subId
+echo
+echo
+
+
+--REGEXPECT--
+01. Create a subscription with encoded chars in custom payload
+==============================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Create an entity to provoke a notification
+==============================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E1?type=none
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. Dump accumulator to ensure the broker correctly decodes the encoded chars in the custom payload
+===================================================================================================
+POST http://127.0.0.1:9997/notify?qs2=a1&qs1=1
+Fiware-Servicepath: /
+Content-Length: 33
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Hs1: 1
+Hs2: a1
+Ngsiv2-Attrsformat: custom
+Host: 127.0.0.1:9997
+Accept: application/json
+Content-Type: application/json; charset=utf-8
```

Comment:
> It should probably be a part of the httpCustom of the subscription ...

Actually, it is include in httpCustom, somehow... I mean, the user can use `headers: {  Content-Type: ... }` to set the right content type. However, if that custom header is not provided, CB should use the "lowest common denominator", which is text/plain.